Add ddev config file (#111)
ddev/ddev.com: README.md

| `npm run textlint`     | Run textlint on content collections                |
| `npm run textlint:fix` | Apply fixable updates to resolve texlint errors    |

### Local Development Setup

1. Run `cp .env.example .env` to create a `.env` file for environment variables. (Don’t check this in!)
2. Create a [classic GitHub access token](https://github.com/settings/tokens) with these scopes: `repo`, `read:org`, `read:user`, and `read:project`.
3. Paste the GitHub token after `.env`’s `GITHUB_TOKEN=`.

> #### DDev setup

Reviewer: ```suggestion
#### DDEV setup
```

> Ddev already has all the dependencies included.

Reviewer: ```suggestion
DDEV already has all the dependencies included.
```

1. Run `ddev start && ddev npm install` set up the project’s dependencies.

> 3. `npm run build` can be found at the base URL and `npm run dev` is found at URL:4321. Dev has Vite HMR(hot module reloading) among other features. The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).

Reviewer: ```suggestion
2. `ddev npm run build` can be found at the base URL and `ddev npm run dev` is found at URL:4321. Dev has Vite HMR(hot module reloading) among other features. The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).
```

Author: ${DDEV_PRIMARY_URL}:4321 -  I was looking for that variable for a while haha, I will work on that.

http://127.0.0.1/ is for the local setup without DDev in case they just want to run npm but maybe I can make that clearer I will double check.

I did not see that build error I will check on that one as well.

Thanks for the feedback.

To generate a static copy of the site, run `ddev npm run build`. The contents of the `dist/` folder are exactly what get [deployed to Cloudflare Pages](#build--deployment). You can preview locally by running `ddev npm run preview` or using a tool like [`serve`](https://www.npmjs.com/package/serve).

#### General setup

Check out the project in your favorite Node.js environment, ideally running [`nvm`](https://github.com/nvm-sh/nvm). We’ll install dependencies, add a GitHub API key, and run a local dev server with a hot-reloading browser URL.

1. Run `nvm use` to make sure you’re running an appropriate Node.js version.
2. Run `npm install` to set up the project’s dependencies.
3. Run `cp .env.example .env` to create a `.env` file for environment variables. (Don’t check this in!)

> 4. Create a [classic GitHub access token](https://github.com/settings/tokens) with these scopes: `repo`, `read:org`, `read:user`, and `read:project`.

Reviewer: Isn't the GITHUB_TOKEN still needed to download the assets?

Author: it is but since we can run it with both ddev and just npm on the host instructions I extracted that and added it as the first few steps when setting up the local environment. So is higher up before ddev setup.

The current order is 
Local development instructions.
- Github token

Then based on that one chooses between the DDev setup or the general Setup.

> 4. Visit the URL displayed in your terminal. (Probably `http://127.0.0.1:4321/`.) The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).

Reviewer: This should be `https://<projectname>.ddev.site:4321`. Maybe 127.0.0.1 works anyway? I doubt it, because then ddev-router doesn't know where to route it to.

Author: This is for the local setup without ddev. Since some might choose that if they have npm already installed.

Reviewer: OK, so I was reading too fast and without adequate context? Please make sure the two sections are well-separated and clear what's what. (They probably are. It's just the problem of reading diffs!)

### Local Development Setup

1. Run `cp .env.example .env` to create a `.env` file for environment variables. (Don’t check this in!)
2. Create a [classic GitHub access token](https://github.com/settings/tokens) with these scopes: `repo`, `read:org`, `read:user`, and `read:project`.
3. Paste the GitHub token after `.env`’s `GITHUB_TOKEN=`.

#### DDev setup

DDev already has all the dependencies included.

1. Run `ddev start && ddev npm install ` set up the project’s dependencies. If it fails then run `ddev npm cache clean --force && ddev npm install && ddev restart`. 
3. `ddev npm run build` can be found at the base URL and `ddev npm run dev` is found at https://<projectname>.ddev.site:4321. Dev has Vite HMR(hot module reloading) among other features. The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).

To generate a static copy of the site, run `ddev npm run build`. The contents of the `dist/` folder are exactly what get [deployed to Cloudflare Pages](#build--deployment). You can preview locally by running `ddev npm run preview` or using a tool like [`serve`](https://www.npmjs.com/package/serve). 

> #### Without DDev setup

Reviewer 2: ```suggestion
#### Setup Without DDEV
```

Check out the project in your favorite Node.js environment, ideally running [`nvm`](https://github.com/nvm-sh/nvm). We’ll install dependencies, add a GitHub API key, and run a local dev server with a hot-reloading browser URL.

1. Run `nvm use` to make sure you’re running an appropriate Node.js version.
2. Run `npm install` to set up the project’s dependencies.
3. Run `npm run dev` to start Astro’s dev server.  If it fails then run `npm cache clean --force && npm install && npm run dev`. 

> 4. Visit the URL displayed in your terminal. (Probably `http://localhost:4321/`.) The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).

Reviewer 2: ```suggestion
4. Visit the URL displayed in your terminal. (Probably `http://localhost:4321/`.) The site will automatically refresh as you work on it, displaying errors in the relevant terminal or browser console.
```

To generate a static copy of the site, run `npm run build`. The contents of the `dist/` folder are exactly what get [deployed to Cloudflare Pages](#build--deployment). You can preview locally by running `npm run preview` or using a tool like [`serve`](https://www.npmjs.com/package/serve).

> #### Switching from without Ddev to with DDev local environment.

Reviewer 2: Use correct “DDEV”s and shorten:

```suggestion
#### Switching from Without DDEV to With DDEV
```

> Make sure to delete your node vendor folder everytime you switch and run npm install. The change in architecture can create odd issues otherwise.

Reviewer 2: Be specific, use backticks for command:

```suggestion
Make sure to delete your `node_modules/` directory and run `npm install`. The change in architecture can create odd issues otherwise.
```

> Make sure to delete your `node_modules/` directory and run `npm install`. The change in architecture can create odd issues otherwise.

Reviewer: (There is no change in architecture though, but probably a difference in node version, related problems.)
```suggestion
Make sure to delete your `node_modules/` directory and run `ddev npm install`. The change in architecture can create odd issues otherwise.
```

Author: True, it could probably be clearer. I meant, npm downloads a different of the files locally as it does inside docker. I thought it was an arm64 vs non arm issue but that does not make much sense now I think about it. But it was something like that.

> DDev already has all the dependencies included.

Reviewer 2: Not sure where “DDev” came from, but it [should always be referred to](https://ddev.readthedocs.io/en/stable/developers/writing-style-guide/#writing-style) as DDEV the product or `ddev` the command.

```suggestion
DDEV already has all the dependencies included.
```

> 1. Run `ddev start && ddev npm cache clean --force  && ddev npm install ` set up the project’s dependencies.

Reviewer 2: Safe to get rid of the extra space unless that was important:

```suggestion
1. Run `ddev start && ddev npm cache clean --force  && ddev npm install` set up the project’s dependencies.
```

> 3. `ddev npm run build` can be found at the base URL and `ddev npm run dev` is found at https://<projectname>.ddev.site:4321. Dev has Vite HMR(hot module reloading) among other features. The site will automatically refresh as you work on it, displaying errors in the relevant console (terminal or browser).

Reviewer 2: 3 → 2, fix spacing, reword slightly for clarity:

```suggestion
2. `ddev npm run build` can be found at the base URL and `ddev npm run dev` is found at https://<projectname>.ddev.site:4321. The dev server has Vite HMR (hot module reloading) among other features. The site will automatically refresh as you work on it, displaying errors in the relevant terminal or browser console.
```

What do you mean by `ddev npm run build` can be found at the base URL?

I know that running the `build` command generates a static version of the site into the `dist/` directory, which you can run the `preview` command to serve, but the wording of this seems sort of ambiguous. Similarly with “`ddev npm run dev` is found at”—the command isn’t found at a browser address, but running it starts a server you can access from the browser at that address. I know you’re aware of this, but the wording leaves it unclear for someone figuring it out.
